### ISSUE TITLE
rename the xarray_utils?

### ISSUE THREAD

**Issue by mathause** [MESMER-group/mesmer#237]
It's a bit a generic, bland name and does not say much. Also @jschwaab prefered something else. Worth reconsidering before v0.9.0

**Comment by mathause** [comment]
See #217 for ideas

**Comment by mathause** [comment]
My current idea is to make them "first level citizens" (not sure if this is the correct term), i.e.:
```python
mesmer.grid
mesmer.mask
mesmer.weighted

mesmer.grid.wrap_to_180(...)
mesmer.mask.mask_ocean(...)
mesmer.weighted.global_mean(...)
```

where I would move the .py files to core and then import from there. I don't like to have them two levels down (`mesmer.process.grid.stack(...)`) and I don't know what a good name for the first level should be anyway.

**Comment by mathause** [comment]
Happy to reconsider if there is a better idea...

**Comment by mathause** [comment]
After #274 I renamed this issue as I am not yet sure that the names are final (i.e. if I am totally happy with them). This concerns:

```python
mesmer.mask.mask_ocean_fraction
mesmer.mask.mask_ocean
mesmer.mask.mask_antarctica

mesmer.globmean.lat_weights
mesmer.globmean.weighted_mean

mesmer.core.computation.calc_geodist_exact
mesmer.core.computation.gaspari_cohn
```

- `calc_geodist_exact`, `gaspari_cohn`, and `lat_weights` could go to `mesmer.geo` or `mesmer.geog`.
- `mesmer.globmean.weighted_mean` could be renamed to `mesmer.weighted.global_mean`. Or this could even go to `mesmer.stats.weighted`?
- `mesmer.mask` could go to `mesmer.region`, or also to `mesmer.region` (I am not super happy having `mask` twice in a row (i.e. `mesmer.mask.mask_ocean_fraction`), but they could just as well stay...


---

I am happy with `mesmer.grid`:
```
mesmer.grid.wrap_to_180(...)
mesmer.grid.wrap_to_360(...)
mesmer.grid.stack_lat_lon(...)
mesmer.grid.unstack_lat_lon_and_align(...)
mesmer.grid.unstack_lat_lon(...)
mesmer.grid.align_to_coords(...)
```

**Comment by mathause** [comment]
- `mesmer.globmean` certainly needs renaming
  - Move this to `mesmer.weighted`
  - Add a new function `global_mean(da, weights=None, x_dim="lon", y_dim="lat")` which calculates the lat weights if not given.
- I recently thought of `geospatial` - so `calc_geodist_exact` could go to `mesmer.geospatial.calc_geodist_exact`
- Need to think if I move e.g. "mesmer/core/grid.py" to "mesmer/grid.py". Organisation-wise I don't like it but sphinx-autosummary does not work well otherwise...

**Comment by mathause** [comment]
closing in favor of #348
